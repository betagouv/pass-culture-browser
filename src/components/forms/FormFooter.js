/* eslint
  react/jsx-one-expression-per-line: 0 */
import PropTypes from 'prop-types'
import React from 'react'
import { Link } from 'react-router-dom'

import { FormFooterObject } from '../../types'

const renderSubmitButton = obj => {
  const props = {
    className: `flex-1 ${obj.className || ''}`,
    disabled: obj.disabled,
  }
  if (obj.id) props.id = obj.id
  return (
    <button
      type="submit"
      {...props}
    >
      <span className="is-block">{obj.label}</span>
    </button>
  )
}

const renderLinkButton = obj => {
  const props = {
    className: `flex-1 ${obj.className || ''}`,
    disabled: obj.disabled,
    to: obj.url,
  }
  if (obj.id) props.id = obj.id
  return (
    <Link {...props}>
      <span className="is-block">{obj.label}</span>
    </Link>
  )
}

export class FormFooter extends React.PureComponent {
  render() {
    const { cancel, className, submit } = this.props
    const useCancel = Boolean(cancel && cancel.url)
    const useLink = Boolean(submit && submit.url)
    const useSubmit = Boolean(submit && !submit.url)
    const hideSeparator = !useCancel || !submit
    return (
      <footer
        className={`pc-final-form-footer dotted-top-2x-white py7 flex-0 flex-columns text-center items-center fs20 ${className}`}
      >
        {useCancel && renderLinkButton(cancel)}
        {!hideSeparator && <hr className="dotted-left-2x-white flex-0" />}
        {useLink && renderLinkButton(submit)}
        {useSubmit && renderSubmitButton(submit)}
      </footer>
    )
  }
}

FormFooter.defaultProps = {
  cancel: false,
  className: '',
  submit: false,
}

FormFooter.propTypes = {
  cancel: PropTypes.oneOfType([PropTypes.bool, FormFooterObject]),
  className: PropTypes.string,
  submit: PropTypes.oneOfType([PropTypes.bool, FormFooterObject]),
}

export default FormFooter
