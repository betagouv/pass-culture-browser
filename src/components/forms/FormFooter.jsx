import PropTypes, { bool, shape, string } from 'prop-types'
import React, { PureComponent } from 'react'
import { Link } from 'react-router-dom'

class FormFooter extends PureComponent {
  componentDidMount() {
    // In Instagram iOS in-app browser, the browser footer hides actions buttons, and it's impossible to scroll
    // this hack replace the actions buttons upper
    const userAgent = navigator.userAgent
    if (userAgent.includes('Instagram') && userAgent.includes('iPhone')) {
      const buttonsContainer = document.getElementById('logout-form-footer')
      buttonsContainer.style.position = 'relative'
      buttonsContainer.style.bottom = '9%'
      const arbitraryValueToScrollToTheBottom = 10000
      window.scrollTo(0, arbitraryValueToScrollToTheBottom)
    }
  }

  renderSubmitButton = options => {
    const attributes = {
      className: `flex-1 ${options.className || ''}`,
      disabled: options.disabled,
      id: options.id,
    }
    return (
      <button
        type="submit"
        {...attributes}
      >
        {options.label}
      </button>
    )
  }

  renderLink = options => {
    const attributes = {
      className: `flex-1 ${options.className || ''}`,
      disabled: options.disabled,
      id: options.id,
      to: options.url,
    }
    return (<Link {...attributes}>
      {options.label}
    </Link>)
  }

  handleTracking = () => {
    if (window.location.href === 'https://app.passculture.beta.gouv.fr/beta') {
      window.gtag_report_conversion(
        'https://www.demarches-simplifiees.fr/commencer/inscription-pass-culture'
      )
    }
  }

  renderExternalLink = options => {
    const attributes = {
      className: `flex-1 ${options.className || ''}`,
      href: options.url,
      id: options.id,
      title: options.title,
    }
    return (
      <a
        {...attributes}
        onClick={this.handleTracking}
        onKeyPress={this.handleTracking}
        role="button"
        tabIndex="0"
        target="_blank"
      >
        {options.label}
      </a>
    )
  }

  render() {
    const { cancel, externalLink, submit } = this.props
    const isCancelLink = Boolean(cancel && cancel.url)
    const isExternalLink = Boolean(externalLink && externalLink.url)
    const isInnerLink = Boolean(submit && submit.url)
    const isSubmitButton = Boolean(submit && !submit.url)
    const hideSeparator = !(isCancelLink || isExternalLink) || !submit

    return (
      <footer
        className="logout-form-footer"
        id="logout-form-footer"
      >
        {isCancelLink && this.renderLink(cancel)}
        {isExternalLink && this.renderExternalLink(externalLink)}
        {!hideSeparator && <hr className="dotted-left-2x-white flex-0" />}
        {isInnerLink && this.renderLink(submit)}
        {isSubmitButton && this.renderSubmitButton(submit)}
      </footer>
    )
  }
}

FormFooter.defaultProps = {
  cancel: null,
  externalLink: null,
  submit: null,
}

FormFooter.propTypes = {
  cancel: shape({
    className: string,
    disabled: bool,
    id: string,
    label: string.isRequired,
    url: string,
  }),
  externalLink: PropTypes.shape({
    className: PropTypes.string,
    id: PropTypes.string,
    label: PropTypes.string.isRequired,
    title: PropTypes.string,
    url: PropTypes.string,
  }),
  submit: shape({
    className: string,
    disabled: bool,
    id: string,
    label: string.isRequired,
    url: string,
  }),
}

export default FormFooter
