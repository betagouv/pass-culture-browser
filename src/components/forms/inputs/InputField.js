import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Field } from 'react-final-form'

import FormError from '../FormError'
import InputLabel from '../InputLabel'
import { validateRequiredField } from '../validators'

class InputField extends Component {
  renderField = ({ input, meta }) => {
    const {
      autoComplete,
      className,
      disabled,
      label,
      name,
      placeholder,
      required,
      sublabel,
    } = this.props

    return (
      <p className={`${className}`}>
        <label
          className="pc-final-form-text"
          htmlFor={name}
        >
          {label && <InputLabel
            label={label}
            required={required}
            sublabel={sublabel}
                    />}
          <span className="pc-final-form-inner">
            <input
              {...input}
              autoComplete={autoComplete ? 'on' : 'off'}
              className="pc-final-form-input is-block"
              disabled={disabled}
              id={name} // cast to boolean
              placeholder={placeholder}
              required={!!required}
              type="text"
            />
          </span>
          <FormError meta={meta} />
        </label>
      </p>
    )
  }

  render() {
    const {
      name,
      required,
    } = this.props
    const validateFunc =
      required && typeof required === 'function'
        ? required
        : (required && validateRequiredField) || undefined
    return (
      <Field
        name={name}
        render={this.renderField}
        validate={validateFunc || undefined}
      />
    )
  }
}

InputField.defaultProps = {
  autoComplete: false,
  className: '',
  disabled: false,
  label: '',
  placeholder: 'Veuillez saisir une valeur',
  required: false,
  sublabel: null,
}

InputField.propTypes = {
  autoComplete: PropTypes.bool,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  label: PropTypes.string,
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  required: PropTypes.oneOfType([PropTypes.bool, PropTypes.func]),
  sublabel: PropTypes.string,
}

export default InputField
