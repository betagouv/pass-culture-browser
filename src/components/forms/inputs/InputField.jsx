import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Field } from 'react-final-form'

import FormError from '../FormError'
import InputLabel from '../InputLabel'
import validateRequiredField from '../validators/validateRequiredField'
import hasErrorMessage from '../utils/hasErrorMessage'

class InputField extends Component {
  renderField = ({ input, meta }) => {
    const {
      autoComplete,
      disabled,
      label,
      name,
      placeholder,
      required,
      sublabel,
      theme,
    } = this.props

    return (
      <label className="label-text-inner">
        <InputLabel
          label={label}
          required={required}
          sublabel={sublabel}
        />
        <div className={`input-inner${hasErrorMessage(meta)}`}>
          <input
            {...input}
            autoComplete={autoComplete ? 'on' : 'off'}
            className="form-input"
            disabled={disabled}
            id={name}
            placeholder={placeholder}
            required={!!required}
            type="text"
          />
        </div>
        <FormError
          meta={meta}
          theme={theme}
        />
      </label>
    )
  }

  render() {
    const { name, required } = this.props
    const validateFunc =
      required && typeof required === 'function'
        ? required
        : (required && validateRequiredField) || undefined

    return (<Field
      name={name}
      render={this.renderField}
      validate={validateFunc || undefined}
            />)
  }
}

InputField.defaultProps = {
  autoComplete: false,
  disabled: false,
  placeholder: 'Veuillez saisir une valeur',
  required: false,
  sublabel: null,
  theme: 'white',
}

InputField.propTypes = {
  autoComplete: PropTypes.bool,
  disabled: PropTypes.bool,
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  required: PropTypes.oneOfType([PropTypes.bool, PropTypes.func]),
  sublabel: PropTypes.string,
  theme: PropTypes.string,
}

export default InputField
