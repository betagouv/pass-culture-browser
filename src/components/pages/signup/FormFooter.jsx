import classnames from 'classnames'
import PropTypes from 'prop-types'
import React from 'react'
import { NavLink } from 'react-router-dom'

const FormFooter = ({ canSubmit, isLoading }) => (
  <footer>
    <button
      className={classnames("button is-primary is-inverted", {
        'is-loading': isLoading,
      })}
      disabled={!canSubmit}
      type="submit"
    >
      Créer
    </button>
    <NavLink to="/connexion" className="button is-secondary">
      {"J'ai déjà un compte"}
    </NavLink>
  </footer>
)

FormFooter.defaultProps = {
  canSubmit: false,
  isLoading: false
}

FormFooter.propTypes = {
  canSubmit: PropTypes.bool,
  isLoading: PropTypes.bool
}

export default FormFooter
