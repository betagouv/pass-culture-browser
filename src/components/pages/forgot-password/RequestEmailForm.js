import React from 'react'
import PropTypes from 'prop-types'

import { FormError, FormFooter } from '../../forms'
import { InputField } from '../../forms/inputs'
import withResetForm from './withResetForm'

const cancelOptions = {
  className: 'is-white-text',
  disabled: false,
  label: 'Annuler',
  url: '/connexion',
}

const submitOptions = {
  className: 'is-bold is-white-text',
  label: 'OK',
}

export const RawRequestEmailForm = ({ canSubmit, isLoading, formErrors }) => (
  <div
    className="is-full-layout flex-rows"
    id="reset-password-page-request"
  >
    <main
      className="pc-main is-white-text flex-1"
      role="main"
    >
      <div className="pc-scroll-container">
        <div className="is-full-layout flex-rows flex-center padded-2x">
          <h2 className="mb36">
            <span className="is-block is-italic is-medium fs22">
              {'Renseignez votre adresse e-mail pour r√©initialiser votre mot de passe.'}
            </span>
            <span className="is-block is-regular fs13 mt18">
              <span>{'*'}</span>
              &nbsp;{'Champs obligatoires'}
            </span>
          </h2>
          <div>
            <InputField
              disabled={isLoading}
              label="Adresse e-mail"
              name="email"
              placeholder="Ex. : nom@domaine.fr"
              required
              theme="primary"
            />
            {formErrors && <FormError customMessage={formErrors} />}
          </div>
        </div>
      </div>
    </main>
    <FormFooter
      cancel={cancelOptions}
      submit={{ ...submitOptions, disabled: !canSubmit }}
    />
  </div>
)

RawRequestEmailForm.defaultProps = {
  formErrors: false,
}

RawRequestEmailForm.propTypes = {
  canSubmit: PropTypes.bool.isRequired,
  formErrors: PropTypes.oneOfType([PropTypes.array, PropTypes.bool, PropTypes.string]),
  isLoading: PropTypes.bool.isRequired,
}

export default withResetForm(RawRequestEmailForm, null, '/users/reset-password', 'POST')
