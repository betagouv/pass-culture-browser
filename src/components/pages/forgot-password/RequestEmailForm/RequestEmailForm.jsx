import PropTypes from 'prop-types'
import React, { Fragment } from 'react'

import FormError from '../../../forms/FormError'
import FormFooter from '../../../forms/FormFooter'
import InputField from '../../../forms/inputs/InputField'
import withNotRequiredLogin from '../../../hocs/with-login/withNotRequiredLogin'
import withResetForm from '../hocs/withResetForm'

export const RequestEmailForm = ({ canSubmit, isLoading, formErrors }) => (
  <Fragment>
    <div>
      <div className="logout-form-header">
        <div className="logout-form-title">
          {'Renseigne ton adresse e-mail pour r√©initialiser ton mot de passe.'}
        </div>
      </div>
      <div>
        <InputField
          disabled={isLoading}
          label="Adresse e-mail"
          name="email"
          placeholder="Ex. : nom@domaine.fr"
          theme="white"
        />
        {formErrors && <FormError customMessage={formErrors} />}
      </div>
    </div>
    <FormFooter
      items={[
        {
          className: 'is-white-text',
          id: 'np-cancel-link',
          label: 'Annuler',
          url: '/connexion',
        },
        {
          className: 'is-bold is-white-text',
          id: 'np-ok-button',
          label: 'OK',
          disabled: !canSubmit,
        },
      ]}
    />
  </Fragment>
)

RequestEmailForm.defaultProps = {
  formErrors: false,
}

RequestEmailForm.propTypes = {
  canSubmit: PropTypes.bool.isRequired,
  formErrors: PropTypes.oneOfType([PropTypes.array, PropTypes.bool, PropTypes.string]),
  isLoading: PropTypes.bool.isRequired,
}

export default withNotRequiredLogin(
  withResetForm(RequestEmailForm, null, '/users/reset-password', 'POST')
)
