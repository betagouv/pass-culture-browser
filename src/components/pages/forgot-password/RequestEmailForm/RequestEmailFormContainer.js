import withPageTracking from '../../../../tracking/withPageTracking'
import RequestEmailForm from './RequestEmailForm'
import { compose } from 'redux'
import withNotRequiredLogin from '../../../hocs/with-login/withNotRequiredLogin'

export default compose(
  withNotRequiredLogin,
  withPageTracking
)(RequestEmailForm)
