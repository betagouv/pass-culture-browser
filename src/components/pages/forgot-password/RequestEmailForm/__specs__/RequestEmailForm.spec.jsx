import { shallow } from 'enzyme'
import React from 'react'

import { RequestEmailForm } from '../RequestEmailForm'
import InputField from '../../../../forms/inputs/InputField'
import FormFooter from '../../../../forms/FormFooter'

describe('src | components | RequestEmailForm', () => {
  it('should display a sentence, an input text field and form footer', () => {
    // given
    const props = {
      canSubmit: true,
      isLoading: false,
    }

    // when
    const wrapper = shallow(<RequestEmailForm {...props} />)

    // then
    const sentence = wrapper.find({
      children: 'Renseigne ton adresse e-mail pour r√©initialiser ton mot de passe.',
    })
    const input = wrapper.find(InputField)
    const formFooter = wrapper.find(FormFooter)
    expect(sentence).toHaveLength(1)
    expect(input).toHaveLength(1)
    expect(formFooter).toHaveLength(1)
  })
})
