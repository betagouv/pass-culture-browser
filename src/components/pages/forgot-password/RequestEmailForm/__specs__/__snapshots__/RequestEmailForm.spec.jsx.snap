// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | forgot-password | RequestEmailForm should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    canSubmit={true}
    formErrors={false}
    isLoading={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div>
          <div
            className="logout-form-header"
          >
            <div
              className="logout-form-title"
            >
              Renseignez votre adresse e-mail pour réinitialiser votre mot de passe.
            </div>
          </div>
          <div>
            <InputField
              autoComplete={false}
              disabled={false}
              label="Adresse e-mail"
              name="email"
              placeholder="Ex. : nom@domaine.fr"
              required={false}
              sublabel={null}
              theme="white"
            />
          </div>
        </div>,
        <FormFooter
          cancel={
            Object {
              "className": "is-white-text",
              "id": "np-cancel-link",
              "label": "Annuler",
              "url": "/connexion",
            }
          }
          externalLink={null}
          submit={
            Object {
              "className": "is-bold is-white-text",
              "disabled": false,
              "id": "np-ok-button",
              "label": "OK",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="logout-form-header"
            >
              <div
                className="logout-form-title"
              >
                Renseignez votre adresse e-mail pour réinitialiser votre mot de passe.
              </div>
            </div>,
            <div>
              <InputField
                autoComplete={false}
                disabled={false}
                label="Adresse e-mail"
                name="email"
                placeholder="Ex. : nom@domaine.fr"
                required={false}
                sublabel={null}
                theme="white"
              />
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="logout-form-title"
              >
                Renseignez votre adresse e-mail pour réinitialiser votre mot de passe.
              </div>,
              "className": "logout-form-header",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Renseignez votre adresse e-mail pour réinitialiser votre mot de passe.",
                "className": "logout-form-title",
              },
              "ref": null,
              "rendered": "Renseignez votre adresse e-mail pour réinitialiser votre mot de passe.",
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <InputField
                  autoComplete={false}
                  disabled={false}
                  label="Adresse e-mail"
                  name="email"
                  placeholder="Ex. : nom@domaine.fr"
                  required={false}
                  sublabel={null}
                  theme="white"
                />,
                false,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "autoComplete": false,
                  "disabled": false,
                  "label": "Adresse e-mail",
                  "name": "email",
                  "placeholder": "Ex. : nom@domaine.fr",
                  "required": false,
                  "sublabel": null,
                  "theme": "white",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              false,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "cancel": Object {
            "className": "is-white-text",
            "id": "np-cancel-link",
            "label": "Annuler",
            "url": "/connexion",
          },
          "externalLink": null,
          "submit": Object {
            "className": "is-bold is-white-text",
            "disabled": false,
            "id": "np-ok-button",
            "label": "OK",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div>
            <div
              className="logout-form-header"
            >
              <div
                className="logout-form-title"
              >
                Renseignez votre adresse e-mail pour réinitialiser votre mot de passe.
              </div>
            </div>
            <div>
              <InputField
                autoComplete={false}
                disabled={false}
                label="Adresse e-mail"
                name="email"
                placeholder="Ex. : nom@domaine.fr"
                required={false}
                sublabel={null}
                theme="white"
              />
            </div>
          </div>,
          <FormFooter
            cancel={
              Object {
                "className": "is-white-text",
                "id": "np-cancel-link",
                "label": "Annuler",
                "url": "/connexion",
              }
            }
            externalLink={null}
            submit={
              Object {
                "className": "is-bold is-white-text",
                "disabled": false,
                "id": "np-ok-button",
                "label": "OK",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="logout-form-header"
              >
                <div
                  className="logout-form-title"
                >
                  Renseignez votre adresse e-mail pour réinitialiser votre mot de passe.
                </div>
              </div>,
              <div>
                <InputField
                  autoComplete={false}
                  disabled={false}
                  label="Adresse e-mail"
                  name="email"
                  placeholder="Ex. : nom@domaine.fr"
                  required={false}
                  sublabel={null}
                  theme="white"
                />
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="logout-form-title"
                >
                  Renseignez votre adresse e-mail pour réinitialiser votre mot de passe.
                </div>,
                "className": "logout-form-header",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Renseignez votre adresse e-mail pour réinitialiser votre mot de passe.",
                  "className": "logout-form-title",
                },
                "ref": null,
                "rendered": "Renseignez votre adresse e-mail pour réinitialiser votre mot de passe.",
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <InputField
                    autoComplete={false}
                    disabled={false}
                    label="Adresse e-mail"
                    name="email"
                    placeholder="Ex. : nom@domaine.fr"
                    required={false}
                    sublabel={null}
                    theme="white"
                  />,
                  false,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "autoComplete": false,
                    "disabled": false,
                    "label": "Adresse e-mail",
                    "name": "email",
                    "placeholder": "Ex. : nom@domaine.fr",
                    "required": false,
                    "sublabel": null,
                    "theme": "white",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                false,
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cancel": Object {
              "className": "is-white-text",
              "id": "np-cancel-link",
              "label": "Annuler",
              "url": "/connexion",
            },
            "externalLink": null,
            "submit": Object {
              "className": "is-bold is-white-text",
              "disabled": false,
              "id": "np-ok-button",
              "label": "OK",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
